src/librbd/operation/SparsifyRequest.cc

```diff
@@ -19,6 +20,62 @@
 namespace librbd {
 namespace operation {
 
+namespace {
+
+bool may_be_trimmed(CephContext *cct,
```

Comment:
I wonder if it would look cleaner if you instead loop through the extents in the extent map, and then loop through sparse-sized aligned chunks within each extent, take a ```subbl.substr_of``` section from the ```bl```, and run the optimized ```is_zero``` method on that chunk. Since you only support removing and trimming for EC, you would do this in reverse order on the extent map.